eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToDbsnpVariantsWrapperProcessor.java

```diff
@@ -63,9 +73,13 @@ public DbsnpVariantsWrapper process(SubSnpNoHgvs subSnpNoHgvs) throws Exception
         List<String> alternateAlleles = subSnpNoHgvs.getAlternateAllelesInForwardStrand();
         for (String alternateAllele : alternateAlleles) {
             SubmittedVariant submittedVariant = subSnpNoHgvsToSubmittedVariant(subSnpNoHgvs, alternateAllele);
-            if (!submittedVariant.isAllelesMatch()) {
-                decluster(subSnpNoHgvs.getSsId(), submittedVariant, operations,
-                          "Declustered: None of the variant alleles match the reference allele");
+            boolean typeMatches = isSameType(clusteredVariant, submittedVariant, subSnpNoHgvs.getDbsnpVariantType());
+            if (!submittedVariant.isAllelesMatch() && !typeMatches) {
+                decluster(subSnpNoHgvs.getSsId(), submittedVariant, operations, DECLUSTERED_ALLELES_AND_TYPE_MISMATCH);
+            } else if (!submittedVariant.isAllelesMatch()) {
+                decluster(subSnpNoHgvs.getSsId(), submittedVariant, operations, DECLUSTERED_ALLELES_MISMATCH);
+            } else if (!typeMatches) {
+                decluster(subSnpNoHgvs.getSsId(), submittedVariant, operations, DECLUSTERED_TYPE_MISMATCH);
```

Comment:
These combinatorial combinations could become problematic if we find more reasons to decluster. What about providing a list of reasons as an argument, then the message could be built prefixing them with a single `Declustered` and concatenating them all?

The most complex output message could look something like this:

> Declustered: None of the variant alleles match the reference allele. The variant type inferred from the alleles does not match the one asserted by dbSNP.